rotkehlchen/api/rest.py

```diff
@@ -3934,10 +3935,21 @@ def _query_kraken_staking_events(
                     entries_missing_prices=entries,
                 )
             # Query events from database
-            events_raw = self.rotkehlchen.data.db.get_history_events(query_filter)
-            events = [StakingEvent.from_history_base_entry(event) for event in events_raw]
+            events_raw, total_entries = self.rotkehlchen.data.db.get_history_events(query_filter)
+            events = []
+            for event in events_raw:
+                try:
+                    staking_event = StakingEvent.from_history_base_entry(event)
+                except DeserializationError as e:
+                    log.warning(f'Found unknown staking event {event}. {str(e)}')
+                    continue
+                events.append(staking_event)
 
-        return {'result': events, 'message': message, 'status_code': HTTPStatus.OK}
+        result = {
+            'events': events,
+            'entries_found': total_entries,
```

Comment:
Are you sure this is what @kelsos asked of you? I bet it's not. Check the existing logic of other api endpoints with pagination.

This is not useful.


Check here: https://github.com/rotki/rotki/blob/e89088b2c1f91f9949555eb74f10c2a3e29c4abf/rotkehlchen/api/rest.py#L878-L913

What the frontend needs is: (1) the number of entries found for this filter ignoring pagination and (2) the total number of entries and (3) a a limit if there is any.

In your case I guess you don't need (3).